backend/src/composite/mod.rs

```diff
@@ -81,17 +104,46 @@ impl<'a, F: FieldElement> Backend<'a, F> for CompositeBackend<'a, F> {
         prev_proof: Option<Proof>,
         witgen_callback: WitgenCallback<F>,
     ) -> Result<Proof, Error> {
-        self.backend_by_machine
-            .get("main")
-            .unwrap()
-            .prove(witness, prev_proof, witgen_callback)
+        if prev_proof.is_some() {
+            unimplemented!();
+        }
+
+        let proof = CompositeProof {
+            proofs: self
+                .per_machine_data
+                .iter()
```

Comment:
Hmm, maybe it would, but I wouldn't do it in this PR, because:
- This allows us to measure the overhead of doing the proofs by machine.
- In the future, we won't be able to prove them in parallel, because we need shared randomness!

Reply:
Makes sense. But in the future, each stage could be run in parallel across all machines, right?

Reply:
I think that could makes sense! Depends how much the prover can parallelize internally though. If it is very parallel, it might even be slower and just increase RAM usage.